api/v1alpha1/argocd_types.go

```diff
@@ -454,7 +454,7 @@ type ArgoCDRepoSpec struct {
 	Version string `json:"version,omitempty"`
 
 	// ExecTimeout specifies the timeout in seconds for tool execution
-	ExecTimeout *int `json:"execTimeout,omitempty"`
+	ExecTimeout string `json:"execTimeout,omitempty"`
```

Comment:
For including the validation, let's define an alias for `string` called `Duration`
```
// +kubebuilder:validation:Pattern:="^(0|(([0-9]+)y)?(([0-9]+)w)?(([0-9]+)d)?(([0-9]+)h)?(([0-9]+)m)?(([0-9]+)s)?(([0-9]+)ms)?)$"
type Duration string
```
Detailed example can be found here. https://github.com/prometheus-operator/prometheus-operator/blob/dff3575c55ee9e6423907e4bfdcb5ac4b8fe9c89/pkg/apis/monitoring/v1/types.go#L45